app/Http/Controllers/CampaignRoleController.php

```diff
@@ -98,6 +121,15 @@ public function store(StoreCampaignRole $request)
         }
         $this->authorize('create', CampaignRole::class);
         $role = CampaignRole::create($request->all());
+        if ($request->has('duplicate') && $request->get('duplicate') != 0) {
+            $oldRole = CampaignRole::where('id', $request->get('role_id'))->first();
+            foreach ($oldRole->permissions as $permission) {
+                /** @var CampaignPermission $newPermission */
+                $newPermission = $permission->replicate(['user_id', 'campaign_id', 'key', 'action', 'table_name', 'entity_type_id', 'access', 'entity_id', 'misc_id']);
```

Comment:
Replicate's first param is a list of configs that are ignored. I'm surprised that this works? Normally it should just be `->replicate(['campaign_role_id'])` 🤔